test/666_abuse_of_start_test.sh

```diff
@@ -8,25 +8,33 @@ weave_on $HOST1 launch
 docker_bridge_ip=$(weave_on $HOST1 docker-bridge-ip)
 proxy_start_container $HOST1 --name=c1
 
-check_hostconfig() {
+check_dns() {
     docker_on $HOST1 attach c2 >/dev/null 2>&1 || true # Wait for container to exit
-    assert "docker_on $HOST1 inspect -f '{{.HostConfig.Dns}} {{.HostConfig.NetworkMode}} {{.State.Running}} {{.State.ExitCode}}' $1" "[$docker_bridge_ip] $2 false 0"
+    assert "docker_on $HOST1 inspect -f '{{.HostConfig.Dns}} {{.State.Running}} {{.State.ExitCode}}' $1" "[$docker_bridge_ip] false 0"
+}
+
+check_networkmode() {
+    docker_on $HOST1 attach c2 >/dev/null 2>&1 || true # Wait for container to exit
+    assert "docker_on $HOST1 inspect -f '{{.HostConfig.NetworkMode}} {{.State.Running}} {{.State.ExitCode}}' $1" "$2 false 0"
 }
 
 # Start c2 with a sneaky HostConfig
 proxy docker_on $HOST1 create --name=c2 $SMALL_IMAGE $CHECK_ETHWE_UP
 proxy docker_api_on $HOST1 POST /containers/c2/start '{"NetworkMode": "container:c1"}'
-check_hostconfig c2 container:c1
+check_dns c2
+check_networkmode c2 container:c1
```

Comment:
It's annoying that this ends up doing two 'attach' and two 'inspect'. I'd just have a single function, with an optional param to indicate whether to check dns.